### PR TITLE
feat(): AM 5496 Controller changes to Include VPN Config as a part of…

### PR DESCRIPTION
**Pull request by rajendra-avesha**
… Slice Config

# Description

Fixes #

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit test cases.

### CODE REVIEW

File: apis/controller/v1alpha1/sliceconfig_types.go [hunk 1]
```diff
@@ -41,7 +41,8 @@ type SliceConfigSpec struct {
 	//+kubebuilder:validation:Minimum=2
 	//+kubebuilder:validation:Maximum=32
 	//+kubebuilder:default:=16
-	MaxClusters int `json:"maxClusters,omitempty"`
+	MaxClusters int               `json:"maxClusters,omitempty"`
+	VPNConfig   *VPNConfiguration `json:"vpnConfig"`
```
Comment (rahulsawra98) [inline apis/controller/v1alpha1/sliceconfig_types.go:45]: lets include `omitempty`

File: service/slice_config_webhook_validation.go [hunk 2]
```diff
@@ -263,6 +263,9 @@ func preventUpdate(ctx context.Context, sc *controllerv1alpha1.SliceConfig, old
 	if sliceConfig.Spec.SliceIpamType != sc.Spec.SliceIpamType {
 		return field.Invalid(field.NewPath("Spec").Child("SliceIpamType"), sc.Spec.SliceIpamType, "cannot be updated")
 	}
+	if sliceConfig.Spec.VPNConfig.Cipher != sc.Spec.VPNConfig.Cipher {
```
Comment (rahulsawra98) [inline service/slice_config_webhook_validation.go:266]: VPNConfig could be nil. It may panic

File: service/worker_slice_gateway_service.go [hunk 3]
```diff
@@ -607,6 +609,18 @@ func (s *WorkerSliceGatewayService) buildMinimumGateway(sourceCluster, destinati
 func (s *WorkerSliceGatewayService) generateCerts(ctx context.Context, sliceName string, namespace string,
 	serverGateway *v1alpha1.WorkerSliceGateway, clientGateway *v1alpha1.WorkerSliceGateway,
 	gatewayAddresses WorkerSliceGatewayNetworkAddresses) error {
+	sliceConfig := &controllerv1alpha1.SliceConfig{}
```
Comment (rahulsawra98) [inline service/worker_slice_gateway_service.go:612]: why is this needed?

File: service/worker_slice_gateway_service.go [hunk 4]
```diff
@@ -626,8 +640,10 @@ func (s *WorkerSliceGatewayService) generateCerts(ctx context.Context, sliceName
 	environment["CLIENT_SLICEGATEWAY_NAME"] = clientGateway.Name
 	environment["SLICE_NAME"] = sliceName
 	environment["CERT_GEN_REQUESTS"], _ = util.EncodeToBase64(&cpr)
+	environment["VPN_CIPHER"] = sliceConfig.Spec.VPNConfig.Cipher
```
Comment (rahulsawra98) [inline service/worker_slice_gateway_service.go:643]: lets add a check if sliceConfig.Spec.VPNConfig != nil

**Comment by rahulsawra98** [comment]
Please add proper description!

**Comment by narmidm** [comment]
If not required please close the PR
